### PR TITLE
CSM: Add client-sided chat commands

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

File: builtin/game/chatcommands.lua [hunk 1]
```diff
@@ -1,37 +1,10 @@
--- Minetest: builtin/chatcommands.lua
+-- Minetest: builtin/client/chatcommands.lua
```
Comment (nerzhul) [inline builtin/game/chatcommands.lua]: wrong comment

**Comment by nerzhul** [comment]
merged thanks

File: builtin/client/chatcommands.lua [hunk 2]
```diff
@@ -0,0 +1,28 @@
+-- Minetest: builtin/client/chatcommands.lua
+
+
+core.register_on_sending_chat_messages(function(message)
+	if not (message:sub(1,1) == "/") then
```
Comment (rubenwardy) [inline builtin/client/chatcommands.lua]: You should share code with the server version

Comment (nerzhul) [inline builtin/client/chatcommands.lua]: @red-001 can you take that in account ?

Comment (red-001) [inline builtin/client/chatcommands.lua]: this uses a few functions that aren't available on the server and it removes parts of the server sided version of this function that aren't need and it also takes  a different number of arguments.

**Comment by paramat** [comment]
@nerzhul @red-001 please link to the 2nd approval.

**Comment by paramat** [comment]
Lacking description.

**Pull request merged**